### PR TITLE
chore: adjust pass SNI in the request chunk

### PR DESCRIPTION
**Pull request by tzssangglass**
Signed-off-by: tzssangglass <tzssangglass@gmail.com>

### CODE REVIEW

File: lib/resty/etcd/v3.lua [hunk 1]
```diff
@@ -679,7 +680,6 @@ local function request_chunk(self, method, path, opts, timeout)
         body    = body,
         query   = query,
         headers = headers,
-        ssl_server_name = self.sni,
```
Comment (spacewander) [inline lib/resty/etcd/v3.lua:682]: Why we don't need SNI in the retry?

File: lib/resty/etcd/v3.lua [hunk 2]
```diff
@@ -593,6 +593,7 @@ local function http_request_chunk(self, http_cli)
         ssl_verify = self.ssl_verify,
         ssl_cert_path = self.ssl_cert_path,
         ssl_key_path = self.ssl_key_path,
+        ssl_server_name = self.sni,
```
Comment (spacewander) [inline lib/resty/etcd/v3.lua:596]: Can we add a test to cover http_request_chunk?

Comment (tzssangglass) [reply lib/resty/etcd/v3.lua:596]: update test case to cover it.

additional notes:

1. in request_uri, if there is no sni, will use host as default sni, see: https://github.com/api7/lua-resty-http/blob/master/lib/resty/http.lua#L916
2. in request_chunk, if there is no sni, it is nil.

And if sni is nil, the sni and certificate host will not be verified.

File: lib/resty/etcd/v3.lua (hunk 1 above)
Comment (tzssangglass) [reply lib/resty/etcd/v3.lua:682]: this is not retry, it is request after connect.

**Pull request merged**